patches/force_cacheable.patch

```diff
@@ -21,7 +21,7 @@ index 89e1319f9..4c037f43a 100644
          }
  
          if (n == 0) {
--            u->cacheable = 0;
+-            u->headers_in.no_cache = 1;
 +            u->cacheable = 1;
```

Comment:
In the old patch it replaced the `u->cacheable` variable. However the new patch removes `u->headers_in.no_cache`, but still adds `u->cacheable`. Is this correct?

Reply:
Old version:
https://github.com/nginx/nginx/blob/release-1.17.8/src/http/ngx_http_upstream.c#L4621-L4660

New version:
https://github.com/nginx/nginx/blob/release-1.25.3/src/http/ngx_http_upstream.c#L4831-L4871

There is this part of code:
```
                if (u->headers_in.no_cache || u->headers_in.expired) {
                    u->cacheable = 0;
                }
```
that will make request not cacheable if `u->headers_in.no_cache = 1` so we need to remove it.

Previously we had:
```
- u->cacheable = 0;
+ u->cacheable = 1;
```
If we don't add `u->cacheable = 1;` this would be the same as:
```
- u->cacheable = 0;
```

Reply:
@lucaspiller updated this file. Now tests are passing.